app/src/main/java/ca/rmen/android/scrumchatter/chart/MemberSpeakingTimePieChart.java

```diff
@@ -85,38 +101,35 @@ static void updateMeetingDateRanges(Context context,
         }
     }
 
-    private static SliceValue createSliceValue(long duration, String memberName) {
+    private static PieChartSlice createPieChartSlice(Context context, long duration, long memberId, String memberName) {
         SliceValue sliceValue = new SliceValue();
         sliceValue.setValue(duration);
         String durationString = DateUtils.formatElapsedTime(duration);
-        String label = String.format("%s###%s", memberName, durationString);
-        sliceValue.setLabel(label);
-        return sliceValue;
+        sliceValue.setLabel(durationString);
+        sliceValue.setColor(ChartUtils.getMemberColor(context, memberId));
+        return new PieChartSlice(sliceValue, memberName);
     }
 
-    private static void setupChart(Context context, PieChartContentBinding pieChartBinding , List<SliceValue> sliceValues) {
+    private static void setupChart(Context context,
+                                   PieChartContentBinding pieChartBinding,
+                                   List<PieChartSlice> pieChartSlices) {
         PieChartData data = new PieChartData();
         data.setHasLabels(true);
         //data.setHasLabelsOutside(true);
 
-        Collections.sort(sliceValues, SLICE_VALUE_COMPARATOR);
-        while (sliceValues.size() > MAX_VALUES) {
-            sliceValues.remove(sliceValues.size() - 1);
+        Collections.sort(pieChartSlices, PIE_CHART_SLICE_COMPARATOR);
+        while (pieChartSlices.size() > MAX_VALUES) {
+            pieChartSlices.remove(pieChartSlices.size() - 1);
         }
 
-        String[] lineColors = context.getResources().getStringArray(R.array.chart_colors);
+        List<SliceValue> sliceValues = new ArrayList<>();
         pieChartBinding.legend.removeAllViews();
-        for (int i = 0; i < sliceValues.size(); i++) {
-            String colorString = lineColors[i % lineColors.length];
-            int color = Color.parseColor(colorString);
-            SliceValue sliceValue = sliceValues.get(i);
-            String label = new String(sliceValue.getLabelAsChars());
-            String memberName = label.substring(0, label.indexOf("###"));
```

Comment:
This parsing with `###` characters was a bit of a hack. Hopefully it's slightly cleaner now with the `PieChartSlice` class.